File: src/main/java/io/github/vertxchina/webverticle/CustomHandler.java
package io.github.vertxchina.webverticle;

import io.vertx.core.Handler;

public class CustomHandler {
  public static <T> Handler<T> handler(Handler<T> handler, Handler<Throwable> exceptionHandler){
    return t -> {
      try{
        handler.handle(t);
      }catch (Throwable e){
        exceptionHandler.handle(e);
      }
    };
  }
}

File: src/main/kotlin/io/github/vertxchina/webverticle/CustomHandler.kt
package io.github.vertxchina.webverticle

fun customHandler(handler: suspend () -> Unit, exceptionHandler: suspend (Throwable) -> Unit): suspend () -> Unit {
  return suspend {
    try {
      handler()
    }catch (throwable:Throwable){
      exceptionHandler(throwable)
    }
  }
}